### ISSUE TITLE
What should we do about deleting entity versions?

### ISSUE THREAD

**Issue by jonc125** [ModellingWebLab/WebLab#39]
In the old Web Lab this removed the version from the DB, and cascaded to remove all linked experiments & their results, thus freeing disk space.

In the new Web Lab versions are more 'implicit' as git commits, and you can't delete one without re-writing the underlying repository history - not a good idea! But we might still want a way of freeing up space by removing linked experiments if it turns out a version is rubbish.

Perhaps a separate 'Delete linked experiments' action for an entity version?

Thoughts @mirams @MichaelClerx ?

Related to #22.

**Comment by MichaelClerx** [comment]
That sounds good!

Only it gets a bit weird if say user A has run a bunch of simulations and user B then decides to delete them all...

**Comment by jonc125** [comment]
Yeah. We might be able to be a bit nicer than the old Web Lab and just delete simulations that you created yourself.

**Comment by MichaelClerx** [comment]
And then would we have to mark the other results as based on a retracted model version?

**Comment by jonc125** [comment]
Not necessarily, because the model version wouldn't actually be gone.

I think we should probably disallow re-writing history entirely. If someone really wants to do that, they should delete the entire entity and start over.

**Comment by jonc125** [comment]
Although we could perhaps use git notes (as in #41) to hide a particular commit from view? But then you could already do that (for others at least) by making it private, so this doesn't get you much extra.

**Comment by mirams** [comment]
Hmmm, I am now banging up against this having broken the APclamp in my latest upload (somehow, yet to be established) and there's no way to revert last commit, remove results, or anything. It wants some way of doing it without deleting the whole entity really.

**Comment by jonc125** [comment]
git lets you add a new commit that reverts previous commits. We could add a 'revert to this version' option when viewing an earlier version of an entity?

**Comment by mirams** [comment]
sounds like a good idea, yeah. That coupled with deleting the whole entity gives you a fair bit of flexibility and prevents complicated git rebasing etc. I suppose? Would still need some magic to remove experiments run with deleted versions.
